strongbox-web-core/src/main/java/org/carlspring/strongbox/interceptors/BaseArtifactControllerInterceptor.java

```diff
@@ -0,0 +1,78 @@
+package org.carlspring.strongbox.interceptors;
+
+import org.carlspring.strongbox.providers.io.RepositoryPathResolver;
+import org.carlspring.strongbox.services.ConfigurationManagementService;
+import org.carlspring.strongbox.storage.Storage;
+import org.carlspring.strongbox.storage.repository.Repository;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.servlet.HandlerMapping;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public class BaseArtifactControllerInterceptor
```

Comment:
@fuss86 I think we should use it within `CustomRequestMappingHandlerMapping` (see `AbstractHandlerMapping.adaptedInterceptors`). It should be similar of how `HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE` works, so this interceptor should check `storageId`+`repositlryId` and if all checks passed then expose this variables as request attributes. These attributes will be used within `LayoutRequestCondition` to match the layout.

Reply:
I like your idea, but:

1. `LayoutRequestCondition#getMatchingCondition` is used to determine the handler/method. Interceptors operate on the handler/method. Your suggestion changes the order of this flow.

2. Even if we try to do it (by not using interceptors) in the `CustomRequestMappingHandlerMapping#getMatchingMapping`, for example (so before  `LayoutRequestCondition#getMatchingCondition` ), then your ` HttpServletRequest` is wrapped by `HandlerMappingIntrospector.RequestAttributeChangeIgnoringWrapper` which has 
```
               @Override
		public void setAttribute(String name, Object value) {
			// Ignore attribute change...
		}
```

so maybe there is a reason Spring is doing it this way ?

Reply:
However, I partially used your suggestion and `RepositoryRequestInterceptor` will be exposing `Repository` instance in the `HttpServletRequest`. It is however not used in `LayoutRequestCondition` as described above.

Reply:
Also, adding this interceptor as `AbstractHandlerMapping` interceptor is a bad choice, since for example `StoragesConfigurationController#addOrUpdateRepository` should not validate existence of the storage and repository.

I'm reverting it back.

Reply:
@fuss86 thanks for having a look at this. I will also check your findings in this branch and get back with result.

Reply:
@fuss86 seems you right about `RequestAttributeChangeIgnoringWrapper` but anyway the `HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE` exposed somehow despite the fact that at some point it goes through  `HandlerMappingIntrospector`. 
Anyway I suggest to work out it same way as Spring MVC does with other request attributes:
1. We should improve the `LayoutRequestCondition` so `getMatchingCondition` method will return new custom `RequestCondition` instance with `storage` and `repository` fields;
2. Override the `handleMatch` method in `CustomRequestMappingHandlerMapping` and expose the `storage` and `repository` variables from `RequestMappingInfo.customConditionHolder` into request attributes (`HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE` exposed within this method as well)
3. Use this attributes from request within `RepositoryRequestInterceptor` to check if requested storage and repository was found and if found check the `inSrvice`.

Also, this way you even don't need the `configurationManagementService` in `RepositoryRequestInterceptor` because it will check that `storage` and `repository` instances should be provided as request attributes, same way as it use provided `pathVariables`.

Reply:
> HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE exposed somehow despite the fact that at some point it goes through HandlerMappingIntrospector.

It is exposed later, I described the situation above

Reply:
it was not clear, sorry :)
please share your thoughts about solution above

Reply:
nevermind, done :)